### ISSUE TITLE
feat(pg_ml): Add pgml.automl() function

### ISSUE THREAD

**Issue by tommichiels** [matroidbe/pg_extensions#32]
## Description

Implement the `pgml.automl()` SQL function for automatic model selection.

## SQL Signature

\`\`\`sql
pgml.automl(
    project_name TEXT,
    relation_name TEXT,
    y_column_name TEXT,
    task TEXT DEFAULT NULL,
    exclude_columns TEXT[] DEFAULT NULL,
    time_limit INTEGER DEFAULT NULL
) RETURNS TABLE (
    project_id BIGINT,
    model_id BIGINT,
    algorithm TEXT,
    task TEXT,
    metrics JSONB,
    deployed BOOLEAN
)
\`\`\`

## Tasks

- [ ] Call PyCaret's `compare_models()` to evaluate 15+ algorithms
- [ ] Return best model based on primary metric
- [ ] Store comparison results for review
- [ ] Auto-deploy best model
- [ ] Support time limit for training

## Algorithms Compared

Classification: lr, knn, nb, dt, rf, et, gbc, xgboost, lightgbm, catboost, ada, svm, ridge, lda, qda

Regression: lr, lasso, ridge, en, dt, rf, et, gbr, xgboost, lightgbm, catboost, ada, knn, huber, br

## Reference

- API doc: `design/pg_ml/api.md`
- PyCaret compare_models: https://pycaret.gitbook.io/docs/get-started/functions/train#compare_models

## Dependencies

- #30 (pgml.train function)

## Acceptance Criteria

- [ ] Compares multiple algorithms
- [ ] Deploys best model automatically
- [ ] Returns metrics for all compared models
